File: src/WhitePad.Server/Models/Messages/JoinRoomResponse.cs
namespace WhitePad.Server.Models.Messages;

public class JoinRoomResponse
{
    public bool Success { get; set; }
    public string? StudentId { get; set; }
    public string? StudentSessionToken { get; set; }
    public string? DisplayName { get; set; }
    public RoomSettings? RoomSettings { get; set; }
    public bool IsLocked { get; set; }
    public bool WaitingRoomEnabled { get; set; }
    public bool WaitingRoomUnlocked { get; set; }
    public string? CurrentQuestion { get; set; }
    public string? Error { get; set; }
}

File: src/WhitePad.iPad/WhitePad.iPad/Models/JoinRoomResponse.swift
import Foundation

struct JoinRoomResponse: Codable, Equatable {
    let success: Bool
    let studentId: String?
    let displayName: String?
    let roomSettings: RoomSettings?
    let isLocked: Bool?
    let waitingRoomEnabled: Bool?
    let waitingRoomUnlocked: Bool?
    let currentQuestion: String?
    let error: String?
}
